mesos-master-detector/src/main/scala/com/mesosphere/mesos/MasterDetector.scala

```diff
@@ -59,13 +59,21 @@ case class Zookeeper(master: String, metrics: Metrics) extends MasterDetector wi
 
   case class ZkUrl(auth: Option[String], servers: String, path: String)
 
+  // This is the JSON format for the Mesos master information saved in Zookeeper.
+  // According to the docs of `setIp`:
+  // The IP address (only IPv4) as a packed 4-bytes integer, stored in network order.
+  //
+  // Scala has no unsigned integer. That means we cannot read eg `3355709612` as int but as long. The `toInt` call
+  // converts the long to a signed integer `-939257684`. Note that we do not care about the number but the bytes.
```

Comment:
Please checkout my Rust example for converting the bytes https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=66105b1f4b3b2d27bb76076dd0629613